src/v/kafka/server/handlers/list_offsets.cc

```diff
@@ -232,9 +233,11 @@ list_offsets_topic(list_offsets_ctx& octx, list_offset_topic& topic) {
         partitions.push_back(std::move(pr));
     }
 
-    return when_all_succeed(partitions.begin(), partitions.end())
+    return ssx::when_all_succeed<
+             chunked_vector<list_offset_partition_response>>(
+             std::move(partitions))
       .then([name = std::move(topic.name)](
-              std::vector<list_offset_partition_response> parts) mutable {
+              chunked_vector<list_offset_partition_response> parts) mutable {
```

Comment:
how many entries were in the vector for it to get an oversized alloc? i wonder if instead of switching to a non-vector version, we should also be limiting parallelism? e.g. did we just spawn off like 10,000 background futures?

Reply:
I was reproducing it as 4000 partitions, which is an overkill.

Every `future<list_offset_partition_response>` is 120b. With warnings starting at 128kb, we could be seeing warnings from this area from a bit more than 1000 partitions.